### PR TITLE
lint cleanup run

### PR DESCRIPTION
**Pull request by NEUDitao**
# Description

Takes care of our linter by either fixing the issues that were mentioned, or excluding the ones that can't be fixed

## Type of change

Please delete options that are not relevant.

- [x] Cleanup

# How Has This Been Tested?

- [x] Manually
- [x] With test suite 

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code where needed 
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/app/components/Nav/NavBar.tsx [hunk 1]
```diff
@@ -116,6 +116,7 @@ export default function NavBar({ courseId }: NavBarProps): ReactElement {
     courseId = profile?.courses[0].course.id;
   }
 
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const [defaultCourse, setDefaultCourse] = useLocalStorage(
```
Comment (NEUDitao) [inline packages/app/components/Nav/NavBar.tsx:120]: `defaultCourse` is unused -- but there's no way to destructure this without an unused variable

File: packages/app/hooks/useProfile.ts [hunk 2]
```diff
@@ -7,8 +7,9 @@ type Hook = () => User;
 
 export const useProfile: Hook = () => {
   const { pathname } = useRouter();
-  const { data, error } = useSWR(`api/v1/profile`, async () =>
-    API.profile.index()
+  const { data, error } = useSWR(
+    `api/v1/profile`,
+    async () => await API.profile.index()
```
Comment (NEUDitao) [inline packages/app/hooks/useProfile.ts:12]: asyncs should have awaits, idk why this one didn't

File: packages/server/src/course/course.entity.ts [hunk 3]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
```
Comment (NEUDitao) [inline packages/server/src/course/course.entity.ts:1]: entities need this, since they have the `(type) => asdfasdfa` decorator on relations

**Review by willstenzel — APPROVED** [review]

**Review by dfarooq610 — APPROVED** [review]

**Review by willstenzel — COMMENTED** [review]
Would we be able to update the husky hook that runs prettier to automatically fix some of lint warnings that you fixed up in here?

**Comment by NEUDitao** [comment]
i want to get around to what will suggested before i merge as a heads up

File: packages/app/components/common/ErrorBoundary.tsx [hunk 4]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
```
Comment (willstenzel) [inline packages/app/components/common/ErrorBoundary.tsx:1]: Could we remove this and add an underscore to the variables, now that the linter rules have been updated?

Comment (NEUDitao) [reply packages/app/components/common/ErrorBoundary.tsx:1]: this is more I'm scared/don't quite know if `componentDidCatch` will play nice if we rename its arguments

**Pull request merged**